addOns/postman/src/main/java/org/zaproxy/addon/postman/PostmanParser.java

```diff
@@ -75,15 +96,216 @@ public void importFromUrl(final String url) throws IllegalArgumentException, IOE
     public void importCollection(String collection) throws JsonProcessingException {
         PostmanCollection postmanCollection = parse(collection);
 
-        // TODO: Extract list of HttpMessage from PostmanCollection and send requests
+        List<HttpMessage> httpMessages = new ArrayList<>();
+        extractHttpMessages(postmanCollection.getItem(), httpMessages);
+
+        requestor.run(httpMessages);
     }
 
     public PostmanCollection parse(String collectionJson) throws JsonProcessingException {
         ObjectMapper objectMapper = new ObjectMapper();
         return objectMapper.readValue(collectionJson, PostmanCollection.class);
     }
 
+    private void extractHttpMessages(List<AbstractItem> items, List<HttpMessage> httpMessages) {
+        for (AbstractItem item : items) {
+            if (item instanceof Item) {
+                HttpMessage httpMessage = extractHttpMessage((Item) item);
+                if (httpMessage != null) {
+                    httpMessages.add(httpMessage);
+                }
+            } else if (item instanceof ItemGroup) {
+                extractHttpMessages(((ItemGroup) item).getItem(), httpMessages);
+            }
+        }
+    }
+
     private static boolean isSupportedScheme(String scheme) {
         return "http".equalsIgnoreCase(scheme) || "https".equalsIgnoreCase(scheme);
     }
+
+    public static boolean isContentTypeAlreadySet(List<KeyValueData> headers) {
+        for (KeyValueData header : headers) {
+            if (header.getKey().equalsIgnoreCase("Content-Type")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static HttpMessage extractHttpMessage(Item item) {
+        Request request = item.getRequest();
+        Url url = request.getUrl();
+
+        if (request == null || url == null) {
+            return null;
+        }
+
+        HttpMessage httpMessage;
+        try {
+            String rawUrl = url.getRaw();
+            httpMessage = new HttpMessage(new URI(rawUrl, false));
```

Comment:
- I'd have expected the raw URL to be escaped/encoded already. Did you check that it isn't?
- The path variables are not being handled yet, add a task?

Reply:
> * I'd have expected the raw URL to be escaped/encoded already. Did you check that it isn't?

No, they are not escaped. Shall I do `URLEncoder.encode(rawUrl, "UTF-8")`?

> * The path variables are not being handled yet, add a task?

Yes, adding a task.

Reply:
Are you sure it's not escaped? The URI class already encodes/escapes but that will lead to incorrect encoding as different characters have different meaning depending on where they are.

Reply:
This is addressed per talk in the meeting, thank you!

Reply:
As discussed in the weekly call, the URLs are getting encoded when imported in ZAP. I am resolving this conversation.